contracts/contracts/automation/ClaimBribesSafeModule.sol

```diff
@@ -0,0 +1,330 @@
+            poolAddress: _poolAddress,
+            rewardContractAddress: _rewardContractAddress,
+            rewardTokens: _getRewardTokenAddresses(_rewardContractAddress)
+        });
+
+        if (bribePoolExists(_poolAddress)) {
+            // Update if it already exists
+            bribePools[bribePoolIndex[_poolAddress]] = bribePool;
+        } else {
+            // If not, Append to the list
+            bribePoolIndex[_poolAddress] = bribePools.length;
+            bribePools.push(bribePool);
+        }
+
+        emit BribePoolAdded(_poolAddress);
+    }
+
+    /**
+     * @dev Update the reward token addresses for all pools
+     */
+    function updateRewardTokenAddresses() external onlyExecutor {
+        BribePoolInfo[] storage _bribePools = bribePools;
+        for (uint256 i = 0; i < _bribePools.length; i++) {
+            BribePoolInfo storage bribePool = _bribePools[i];
+            bribePool.rewardTokens = _getRewardTokenAddresses(
+                bribePool.rewardContractAddress
+            );
+        }
+    }
+
+    /**
+     * @dev Get the reward token addresses for a given reward contract address
+     * @param _rewardContractAddress The address of the reward contract
+     * @return _rewardTokens The reward token addresses
+     */
+    function _getRewardTokenAddresses(address _rewardContractAddress)
+        internal
+        view
+        returns (address[] memory)
+    {
+        address[] memory _rewardTokens = new address[](
+            ICLRewardContract(_rewardContractAddress).rewardsListLength()
+        );
+        for (uint256 i = 0; i < _rewardTokens.length; i++) {
+            _rewardTokens[i] = ICLRewardContract(_rewardContractAddress)
+                .rewards(i);
+        }
+
+        return _rewardTokens;
+    }
+
+    /**
+     * @dev Remove a bribe pool from the list
+     * @param _poolAddress The address of the pool to remove
+     */
+    function removeBribePool(address _poolAddress) external onlySafe {
+        if (!bribePoolExists(_poolAddress)) {
+            // If it doesn't exist, skip
+            return;
+        }
+
+        uint256 index = bribePoolIndex[_poolAddress];
+        BribePoolInfo memory lastBribePool = bribePools[bribePools.length - 1];
+        bribePools[index] = lastBribePool;
+        bribePoolIndex[lastBribePool.poolAddress] = index;
+        bribePools.pop();
+
+        emit BribePoolRemoved(_poolAddress);
+    }
+
+    /**
+     * @dev Check if a bribe pool exists
+     * @param bribePool The address of the pool to check
+     * @return true if the pool exists, false otherwise
+     */
+    function bribePoolExists(address bribePool) public view returns (bool) {
+        BribePoolInfo[] memory _bribePools = bribePools;
+        uint256 poolIndex = bribePoolIndex[bribePool];
+        if (poolIndex >= _bribePools.length) {
+            return false;
+        }
+
+        return _bribePools[poolIndex].poolAddress == bribePool;
```

Comment:
nit: Do you think it could be possible to have the same code structure as there is for `nftIdExists()`? 
What I mean is that it could be nice to have all the value returned "calculated" on the return.

Reply:
Sure, will do